src/charm.py

```diff
@@ -138,6 +141,27 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
         event.set_results(results)  # type: ignore[arg-type]
         self.model.unit.status = ops.ActiveStatus()
 
+    def _on_register_user_action(self, event: ActionEvent) -> None:
+        """Reset instance and report action result.
+
+        Args:
+            event: Event triggering the reset instance action.
+        """
+        results = {
+            "register-user": False,
+        }
+        username = event.params["username"]
+        password = event.params["password"]
+        admin = bool(event.params["admin"] == "yes")
+        try:
+            self._synapse_api.register_user(username=username, password=password, admin=admin)
+            results["register-user"] = True
```

Comment:
Should this be together with `event.set_results`

Reply:
Could you give me an example, please?
Do you mean something like this?
```
event.set_results({register-user": True, "user-password": password})
```

Reply:
As in, this line is within the `try ... except` but `event.set_result` isn't, shouldn't both be either in the `try ... except` or both out?

Reply:
Oh, I get it, you're right, it doesn't make sense to be inside the try. I'll move it out.